runtime/interface.go

```diff
@@ -41,93 +43,124 @@ type Block struct {
-	SetValue(owner, key, value []byte) (err error)
-	// CreateAccount creates a new account.
-	CreateAccount(payer Address) (address Address, err error)
-	// AddAccountKey appends a key to an account.
-	AddAccountKey(address Address, publicKey []byte) error
-	// RemoveAccountKey removes a key from an account by index.
-	RemoveAccountKey(address Address, index int) (publicKey []byte, err error)
+type AddressLocation = common.AddressLocation
+
+// Accounts manages changes to the accounts, account keys and account storage
+//
+// errors returns by methods are non-fatal ones (e.g. accountNotExist),
+// fatal errors (e.g. Storage failure) are called by panic inside the methods
+// methods with power of edits have caller as an argument which can be used
+// to limit access to these method calls from smart contracts only, transaction only, or
+// specific smart contracts (e.g. service account).
+type Accounts interface {
+
+	// NewAccount creates a new account address and set the exists flag for this account
+	NewAccount(caller Location) (address Address, err error)
+	// Exists returns true if the account exists
+	Exists(address Address) (exists bool, err error)
+	// NumberOfAccounts returns the number of accounts
+	NumberOfAccounts(caller Address) (count uint64, err error)
+
+	// SuspendAccount suspends an account (set suspend flag to true)
+	SuspendAccount(address Address, caller Location) error
+	// UnsuspendAccount unsuspend an account (set suspend flag to false)
+	UnsuspendAccount(address Address, caller Location) error
+	// returns true if account is suspended
+	IsSuspended(address Address) (isSuspended bool, err error)
+
+	// TODO ramtin merge this with 	Code(location Location) error
+	// AccountContractCode returns the code associated with an account contract.
+	ContractCode(address AddressLocation) (code []byte, err error)
 	// UpdateAccountContractCode updates the code associated with an account contract.
-	UpdateAccountContractCode(address Address, name string, code []byte) (err error)
-	// GetAccountContractCode returns the code associated with an account contract.
-	GetAccountContractCode(address Address, name string) (code []byte, err error)
-	// RemoveAccountContractCode removes the code associated with an account contract.
-	RemoveAccountContractCode(address Address, name string) (err error)
-	// GetSigningAccounts returns the signing accounts.
-	GetSigningAccounts() ([]Address, error)
-	// Log logs a string.
-	Log(string) error
-	// EmitEvent is called when an event is emitted by the runtime.
-	EmitEvent(cadence.Event) error
+	UpdateContractCode(address AddressLocation, code []byte, caller Location) (err error)
+	// RemoveContractCode removes the code associated with an account contract.
+	RemoveContractCode(address AddressLocation, caller Location) (err error)
+	// Contracts returns a list of contract names under this account
+	Contracts(address AddressLocation, caller Location) (Name []string, err error)
```

Comment:
What is this for?

Reply:
returns the list of contracts stored on this account, I saw an internal method call inside ResolveLocation thought might be useful for otherplaces as well.